File: modules/alexa/platform/src/SpeechSynthesizer.cpp
/*
 * Copyright 2017-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

#include <AACE/Alexa/SpeechSynthesizer.h>

namespace aace {
namespace alexa {

SpeechSynthesizer::~SpeechSynthesizer() = default;  // key function
#ifdef OBIGO_AIDAEMON
bool SpeechSynthesizer::startTTS(std::string startEvent, std::string finishEvent) {
  if (auto m_speechSynthesizerEngineInterface_lock = m_speechSynthesizerEngineInterface.lock()) {
    return m_speechSynthesizerEngineInterface_lock->onstartTTS(startEvent, finishEvent);
  } else {
    return false;
  }
}

void SpeechSynthesizer::setEngineInterface(
    std::shared_ptr<aace::alexa::SpeechSynthesizerEngineInterface> speechSynthesizerEngineInterface) {
  m_speechSynthesizerEngineInterface = speechSynthesizerEngineInterface;
}
#endif
}  // namespace alexa
}  // namespace aace

File: platforms/android/modules/alexa/src/main/java/com/amazon/aace/alexa/SpeechSynthesizer.java
/*
 * Copyright 2017-2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

package com.amazon.aace.alexa;

import com.amazon.aace.core.PlatformInterface;

/**
 * SpeechSynthesizer should be extended to handle Alexa speech events from the Engine.
 *
 * @note For observing Alexa dialog state transitions, see @c AlexaClient.dialogStateChanged().
 */
public class SpeechSynthesizer extends PlatformInterface {
    // NativeRef implementation
    final protected long createNativeRef() {
        return createBinder();
    }

    final protected void disposeNativeRef(long nativeRef) {
        disposeBinder(nativeRef);
    }

    // Native Engine JNI methods
    private native long createBinder();
    private native void disposeBinder(long nativeRef);
}
